src/Gui/Tree.cpp

```diff
@@ -855,19 +860,35 @@ void TreeWidget::contextMenuEvent (QContextMenuEvent * e)
         DocumentObjectItem* objitem = static_cast<DocumentObjectItem*>
             (this->contextItem);
 
+        // check that the selection is not across several documents
+        bool acrossDocuments = false;
+        auto SelectedObjectsList = Selection().getCompleteSelection();
+        QList<QTreeWidgetItem*> items = this->selectedItems();
+        // get the object's document as reference
         App::Document* doc = objitem->object()->getObject()->getDocument();
+        for (auto it = SelectedObjectsList.begin(); it != SelectedObjectsList.end(); ++it) {
+            if ((*it).pDoc != doc)
+                acrossDocuments = true;
+        }
+                
         showHiddenAction->setChecked(doc->ShowHidden.getValue());
         contextMenu.addAction(this->showHiddenAction);
 
         hideInTreeAction->setChecked(!objitem->object()->showInTree());
         contextMenu.addAction(this->hideInTreeAction);
 
-        if (objitem->object()->getObject()->isDerivedFrom(App::DocumentObjectGroup::getClassTypeId()))
-            contextMenu.addAction(this->createGroupAction);
+        if (!acrossDocuments) { // is only sensible for selections within one document
+            if (objitem->object()->getObject()->isDerivedFrom(App::DocumentObjectGroup::getClassTypeId()))
+                contextMenu.addAction(this->createGroupAction);
+            contextMenu.addAction(this->selectAllChildrenAction);
+        }
 
         contextMenu.addAction(this->markRecomputeAction);
         contextMenu.addAction(this->recomputeObjectAction);
-        contextMenu.addAction(this->relabelObjectAction);
+
+        // relabeling is only possible for a single selected document
+        if (SelectedObjectsList.size() == 1)

```

Comment:
> fix bug that when more than one object is selected, you get a context menu to rename, despite it is unclear what object should be renamed

Look at onRelabelObject(): this function does not rename a selected item but the current item (if there is one). So, when several items are selected then usually it's the last selected one. Insofar it's quite clear what object will be renamed.

Reply:
This confused me:
* I want to rename one item, so I have to select this one. This works.
* I have to rename several items, so I select them, choose 'Rename' but can still only rename one.

That's why I changed it.

Here I select 3 objects and I can select in the context menu to rename. But it depends on where I click in the selection what will be renamed:
![AnJeJ1Q2kl](https://user-images.githubusercontent.com/1828501/102156739-75e9f380-3e7e-11eb-8121-7582205be4fd.gif)

It took me a while to understand this behavior and I don't see the benefit of having several items selected when I can only rename one per step.
(By the way, a useful feature to rename several objects at once would for example be to select all sketches and give it the same prefix or to change its common prefix.)

Reply:
(BTW To rename several objects at once, one just needs to edit their Label property). But maybe the rename option could be grayed out if more than one object is selected?